### PR TITLE
ISDK-2976: Add arm64 iPhone simulator support

### PR DESCRIPTION
**Pull request by paynerc**
This PR adds iPhone Simulator support for the M1 (arm64) based Macs.

During the investigation, I was unable to get `container` to build successfully for the `arm64` iPhone simulator. We are not using this for Voice or Video, so I removed it. I am not sure if this is necessary for the other consumers of this artifact.

Some of these changes are brought forth from the repo where this work was initially based, https://github.com/faithfracture/Apple-Boost-BuildScript.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by mmartikainen — APPROVED** [review]
Great job on this, looking forward to trying it!
Next: arm64 support for darwin..? :-)

**Comment by paynerc** [comment]
> Do you still need to support 32 bits iphones?

Unfortunately for the time being, yes... yes we do. There is an ongoing discussion with a very large customer about adding support back all the way to iOS 9. So while that is still pending, we need to ensure that we can deliver all the dependencies if we end up having to support this customer.

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,6 +78,9 @@ commands:
           command: |
             if [ -z "$CIRCLE_TAG" ]; then
               echo "Must use release tag to build boost releases. Push \`release-1.71.0-twilio3\`-style tag to trigger."
+              # Remove this 
+              echo 'export BOOST_VERSION=1.73.0' >> $BASH_ENV
```
Comment (dmorilha-twilio) [inline .circleci/config.yml]: please remove it :-)

**Review by dmorilha-twilio — APPROVED** [review]

**Review by mmartikainen — APPROVED** [review]
Awsome! 👍

**Pull request merged**